lib/diagnostic.rb

```diff
@@ -5,29 +5,29 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.sort { |a, b| b.length <=> a.length }
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.reduce(:+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.odd?.reduce(:+)
   end
 
```

Comment:
This isn't going to do much other than return false (because of `numbers.odd?`)

```rb
  def some_odd(numbers)
    # Your code goes here
    numbers.any?(&:odd?)
  end
```